src/dataset/DatasetTree.ts

```diff
@@ -250,6 +250,19 @@ export class DatasetTree extends ZoweTreeProvider implements IZoweTree<IZoweData
         this.refresh();
     }
 
+    public async editSession(node: IZoweDatasetTreeNode) {
```

Comment:
Duplication of code?
I have not read the entire PR, but would it be possible to consolidate these functions a bit more and if they are the same across all node, maybe place them in `shared/utils` ?

Reply:
The approach was patterned to addSession() where there is an editSession function in IZoweTree just so I could identify the node type. Then it will be transferred to each tree. If I put this to utils, I still need the nodes from different trees so it will still be 3 different functions.

Reply:
This should be pushed down into ZoweTreeProvider as editSession(node: IZoweTreeNode) you can also put setProfile and setSession here too. 
My original intention with interfaces and OO approach to Zowe Explorer is that if we can lock down the classes by making them more complete rather than have a proliferation of files and functions we can have explicit interfaces for nodes and trees that can be exported as the extension points for extenders

Reply:
I didn't know that there is a ZoweTreeProvider. I patterned this to addSession. If I transfer this today, I don't think I will be able to make it to 1.5 release. Is it okay to add it in my next profile story? I can refactor it there.

Reply:
Just did the above and it works fine if you want me to show you how it looks I'll post the code

Reply:
Also the refreshTree in utils.ts doesn't seem to be used at all. Can this be removed?.

Reply:
Created a PR to your Branch to show how easy it is ;-)
https://github.com/zowe/vscode-extension-for-zowe/pull/728

Reply:
By the way AddSession has not been updated. I have been in the process of making these changes top clean up the interface but have not been able to complete because the amount of other code restructuring happening at the same time. rename is an example which has been completed

Reply:
@Colin-Stone RefreshTree is used in refreshAll*** functions

Reply:
Thank you for showing me the code. I updated the PR :)